ERCS/erc-7579.md

```diff
@@ -267,15 +315,23 @@ Fallback handlers that implement authorization control, MUST NOT rely on `msg.se
 Hooks MUST implement the `IModule` and the `IHook` interface and have module type id: `4`.
 
 ```solidity
-interface IHook {
+interface IHook is IModule {
     /**
-    *  MAY return arbitrary data in the `hookData` return value
-    */
-    function preCheck(address msgSender, bytes calldata msgData) external returns (bytes memory hookData);
+     * @dev Called by the smart account before execution
+     * @param msgSender the address that called the smart account
+     * @param msgValue the value that was sent to the smart account
+     * @param msgData the data that was sent to the smart account
+     *
+     * MAY return arbitrary data in the `hookData` return value
+     */
+    function preCheck(address msgSender, uint256 msgValue, bytes calldata msgData) external returns (bytes memory hookData);
 
     /**
-    * MAY validate the `hookData` to validate transaction context of the `preCheck` function
-    */
+     * @dev Called by the smart account after execution
+     * @param hookData the data that was returned by the `preCheck` function
+     *
+     * MAY validate the `hookData` to validate transaction context of the `preCheck` function
+     */
     function postCheck(bytes calldata hookData) external returns (bool success);
```

Comment:
Would it be better to also pass **msgSender** and **msgValue** in `postCheck()`? If the hook requires this data, which I think is quite likely, it'll be cheaper than having to write it all into storage between the pre and post checks.

Reply:
this would be in memory - but you think it would be better to pass sender, value, data and return of preCheck?